File: src/main/java/social/alone/server/auth/email/AuthTokenGenerator.java
package social.alone.server.auth.email;

import lombok.RequiredArgsConstructor;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Service;
import social.alone.server.auth.oauth2.user.TokenProvider;

@Service
@RequiredArgsConstructor
public class AuthTokenGenerator {

    private final AuthenticationManager authenticationManager;

    private final TokenProvider tokenProvider;

    public String byEmailPassword(String email, String password) {
        Authentication authentication = authenticationManager.authenticate(
                new UsernamePasswordAuthenticationToken(email, password)
        );

        String token = tokenProvider.createToken(authentication);

        SecurityContextHolder.getContext().setAuthentication(authentication);

        return token;
    }
}

File: src/main/kotlin/social/alone/server/auth/AuthTokenGenerator.kt
package social.alone.server.auth

import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Service
import social.alone.server.auth.oauth2.user.TokenProvider
import social.alone.server.user.domain.User

@Service
class AuthTokenGenerator(val tokenProvider: TokenProvider, val authenticationManager: AuthenticationManager) {

    fun byEmailPassword(email: String, password: String): String {
        val authentication = authenticationManager!!.authenticate(
                UsernamePasswordAuthenticationToken(email, password)
        )

        val token = tokenProvider!!.createToken(authentication)
        SecurityContextHolder.getContext().authentication = authentication

        return token
    }

    fun byUser(user: User): String {
        return tokenProvider!!.createToken(user)
    }
}
